csm_web/frontend/src/components/Course.js

```diff
@@ -0,0 +1,180 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { fetchJSON, fetchWithMethod, HTTP_METHODS } from "../utils/api";
+import LocationIcon from "../../static/frontend/img/location.svg";
+import UserIcon from "../../static/frontend/img/user.svg";
+import GroupIcon from "../../static/frontend/img/group.svg";
+import ClockIcon from "../../static/frontend/img/clock.svg";
+import CheckCircle from "../../static/frontend/img/check_circle.svg";
+import XCircle from "../../static/frontend/img/x_circle.svg";
+import Modal from "./Modal";
+
+const DAY_OF_WEEK_ABREVIATIONS = Object.freeze({
+  Mon: "M",
+  Tue: "Tu",
+  Wed: "W",
+  Thu: "Th",
+  Fri: "F",
+  Sat: "Sa",
+  Sun: "Su"
+});
+
+export default class Course extends React.Component {
+  state = { sections: null, loaded: false, day: "", showUnavailable: true }; // Sections are grouped by day
+
+  static propTypes = {
+    match: PropTypes.shape({ params: PropTypes.shape({ id: PropTypes.string.isRequired }) }).isRequired,
+    name: PropTypes.string.isRequired
+  };
+
+  componentDidMount() {
+    const { id } = this.props.match.params;
+    fetchJSON(`/courses/${id}/sections`).then(sections =>
+      this.setState({ sections, loaded: true, day: Object.keys(sections)[0] })
+    );
+  }
+
+  render() {
+    const { loaded, sections, day: currDay } = this.state;
+    return !loaded ? null : (
+      <div id="course-section-selector">
+        <div id="course-section-controls">
+          <h2 className="course-title">{this.props.name}</h2>
+          <div id="day-selector">
+            {Object.keys(sections).map(day => (
+              <button
+                className={`day-btn ${day == currDay ? "active" : ""}`}
+                key={day}
+                onClick={() => this.setState({ day })}
+              >
+                {DAY_OF_WEEK_ABREVIATIONS[day]}
+              </button>
+            ))}
+          </div>
+          <label id="show-unavailable-toggle">
+            <input
+              type="checkbox"
+              checked={this.state.showUnavailable}
+              onChange={({ target: { checked } }) => this.setState({ showUnavailable: checked })}
+            />
+            Show unavailable
+          </label>
+        </div>
+        <div id="course-section-list">
+          {sections[currDay]
+            .filter(({ numStudentsEnrolled, capacity }) => this.state.showUnavailable || numStudentsEnrolled < capacity)
+            .map(section => (
+              <SectionCard key={section.id} {...section} />
```

Comment:
As a quality of life thing, consider displaying a card that says "No available sections" or similar to make sure that the day of week selector doesn't change place if no sections are available.

Reply:
Added, good idea